apps/file-integrity-monitor/agent/fim-agent/fim-agent.py

```
@@ -0,0 +1,547 @@
+def create_json(data, file_name):
+    os.makedirs(json_target_dir, exist_ok=True)
+    full_file_path = os.path.join(json_target_dir, file_name)
+    json_data = json.dumps(data, indent=4)
+    with open(full_file_path, 'w') as f:
+        f.write(json_data)
+
+
+def is_readable_text_file(file_path):
+
+    try:
+        result = subprocess.run(['/usr/bin/file', file_path], capture_output=True, text=True)
```

Ruff (0.15.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)